js/main.js

```diff
@@ -1,22 +1,91 @@
-function getRandomInteger(min, max){
-  min = min < 0 ? 0 : Math.floor(min);
-  max = max > 0 ? Math.floor(max) : 0;
-  if(min < max && max !== 0) {
-    return Math.floor(Math.random() * (max - min + 1) + min);// источник - https://learn.javascript.ru/number
-  }
-  throw new Error('getRandomInteger — максимальное число не может равняться нулю и должно быть больше минимального, значения float округляются до ближайшего целого');
-}
+/* Функция взята из интернета и доработана, Источник - https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_random */
+
+const LODGINGS = ['palace', 'flat', 'house', 'bungalow', 'hotel'];
+
+const ADS_COUNT = 10;
+
+const getRandomPositiveInteger = function (numValue, otherNumValue) {
+  const lower = Math.ceil(Math.min(Math.abs(numValue), Math.abs(otherNumValue)));
+  const upper = Math.floor(Math.max(Math.abs(numValue), Math.abs(otherNumValue)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
+};
+
+const getRandomPositiveFloat = function (numValue, otherNumValue, digits = 1) {
+  const lower = Math.min(Math.abs(numValue), Math.abs(otherNumValue));
+  const upper = Math.max(Math.abs(numValue), Math.abs(otherNumValue));
+  const result = Math.random() * (upper - lower) + lower;
+  return result.toFixed(digits);
+};
 
-getRandomInteger(5, 50);
+const getAuthor = function () {
+  const authorPictureLink = `img/avatars/user0${getRandomPositiveInteger(1, 8)}.png`;
+  return {
+    avatar: authorPictureLink,
+  };
```

Comment:
Каждое создание переменной занимает место в памяти, пусть даже временно. В своем коде я придерживаюсь правила: **не создавай переменную, если она понадобится тебе всего один раз**. Можно аппелировать к тому, что создание одноразовой переменной улучшает чтение кода и иногда это правда так, когда, скажем, у нас есть какие-то очень сложные калькуляции и имя переменной вносит ясность в происходящее. 
Но в 80% случаев такие переменные наоборот ухуджают читаемость и неоправданно увеличивают количество кода, потому как указывают на и без того очевидные конструкции. Приглядывай за такими кейсами)

```suggestion
  return {
    avatar: `img/avatars/user0${getRandomPositiveInteger(1, 8)}.png`,
  };
```